### ISSUE TITLE
"Avoid unused local variables" when assigning within an array

### ISSUE THREAD

**Issue by gregrobson** [phpmd/phpmd#239]
I'm getting a false positive with the following example:

```php
public function foobar()
{
	extract($row);
	$result = [
		'value' => $valueIsUsed,
	];
}
```

*(background - I'm extracting a database row result and assigning into a more convenient form)*

Is it possible for `$valueIsUsed` to be detected as it's on the "right hand side" of an assignment.

**Comment by klapifoch** [comment]
Same issue here!

**Comment by ravage84** [comment]
@gregrobson Assuming you have one little error in your example:

```` php
public function foobar($row) // Missed parameter here
{
    extract($row);
    $result = [
        'value' => $valueIsUsed,
    ];
}
````
In this case it can't be detected as the content of the ``$row`` variable is dependent on what it is given at runtime.

Same goes for:
```` php
public function foobar()
{
    $row = getRow(); // Get data from function/method
    extract($row);
    $result = [
        'value' => $valueIsUsed,
    ];
}
````
But it could be detected in the following case, as the content of ``$row`` variable is set at "compile time".
```` php
public function foobar()
{
    $row = ['valueIsUsed' => true];
    extract($row);
    $result = [
        'value' => $valueIsUsed,
    ];
}
````
> Is it possible for $valueIsUsed to be detected as it's on the "right hand side" of an assignment.

That does not make any difference and creating some logic for detecting that would be a) relatively extensive b) error prone c) not really worth the problem.

I can relate to your problem as frameworks and ORM libraries tend to have uses of ``extract()`` quite often. But using ``extract()`` makes the code not really readable anyway... so I guess there is no real solution when it comes to phpmd in this case.

Feel free to open a PR with a potential solution, if you find one.
